### PR TITLE
Update generate_build_matrix.py

### PR DESCRIPTION
**Pull request by mablue**
torch 2.5.0 to 2.5.1

### CODE REVIEW

File: scripts/github_actions/generate_build_matrix.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def generate_build_matrix(
                 else ["11.8.0", "12.1.0", "12.4.0"]
             ),
         },
-        "2.5.0": {
+        "2.5.1": {
```
Comment (csukuangfj) [inline scripts/github_actions/generate_build_matrix.py:275]: Could you create a new entry for `2.5.1` and keep the original `2.5.0`?

File: scripts/github_actions/generate_build_matrix.py [hunk 2]
```diff
@@ -284,12 +284,12 @@ def generate_build_matrix(
...
     }
     if test_only_latest_torch:
-        latest = "2.5.0"
+        latest = "2.5.1"
         matrix = {latest: matrix[latest]}
 
     if for_windows or for_macos:
-        if "2.5.0" in matrix:
-            matrix["2.5.0"]["python-version"].remove("3.13")
+        if "2.5.1" in matrix:
```
Comment (csukuangfj) [inline scripts/github_actions/generate_build_matrix.py:291]: Please keep the original `2.5.0` and you can add a new test for `2.5.1`

**Review by csukuangfj — CHANGES_REQUESTED** [review]

File: scripts/github_actions/generate_build_matrix.py (hunk 2 above)
Comment (mablue) [reply scripts/github_actions/generate_build_matrix.py]: done

File: scripts/github_actions/generate_build_matrix.py (hunk 1 above)
Comment (mablue) [reply scripts/github_actions/generate_build_matrix.py:284]: done

File: scripts/github_actions/generate_build_matrix.py [hunk 3]
```diff
@@ -281,10 +281,19 @@ def generate_build_matrix(
...
+                ["11.8", "12.1", "12.4"]  # default 12.4
+                if not for_windows
+                else ["11.8.0", "12.1.0", "12.4.0"]
+            ),
+        },
         # https://github.com/Jimver/cuda-toolkit/blob/master/src/links/windows-links.ts
     }
     if test_only_latest_torch:
-        latest = "2.5.0"
+        latest = "2.5.1"
```
Comment (csukuangfj) [inline scripts/github_actions/generate_build_matrix.py:296]: Please also update the if statement at line 299 to test for `2.5.1`.

Comment (mablue) [reply scripts/github_actions/generate_build_matrix.py:296]: I confused.but I appended a new condition.Are these changes correct?

```
+ if "2.5.1" in matrix:
+            matrix["2.5.1"]["python-version"].remove("3.13")
+
if "2.5.0" in matrix:
            matrix["2.5.0"]["python-version"].remove("3.13")

```

**Comment by csukuangfj** [comment]
Thank you for your contribution!

**Pull request merged**